### DISCUSSION TITLE
number and highlight ALL search results?

### DISCUSSION THREAD

**Discussion by OogieBoogie93** [sumatrapdfreader/sumatrapdf#3594] [Q&A]
Hello!
im new to Sumatra, and when i use the search function it doesn't say the total number of results (usually pdf readers show the current results is X out of Y results) is there an option to add this function? additionally, is there a way to highlight the all the search results? for example, if I search the word "home" which appear 3 times in the same line, I expect it do highlight all of them.
thanks everyone!

**Comment by GitHubRulesOK** [comment]
Text search result is nearest 1 (the current match, biased forwards) all others are unknown as it is reel time match per underlying focus. 

No search ahead, no search in reverse. Thus no known number of matches especially as unknown choice of next character, it is not historic nor predictive.

It should be slower in say Acrobat as it will store the whole file contents as predictive indexed contents.

Best example is FireFox where all the PDF has been converted then buffered, so all is available for matching by numbers, but I dont think it does that instantly it is just part of slower converting, the whole file into images and HTML.
